src/librustc_mir/borrow_check/path_utils.rs

```diff
@@ -133,20 +133,13 @@ pub(super) fn is_active<'tcx>(
 /// Determines if a given borrow is borrowing local data
 /// This is called for all Yield statements on movable generators
 pub(super) fn borrow_of_local_data(place: &Place<'_>) -> bool {
-    place.iterate(|place_base, place_projection| {
-        match place_base {
-            PlaceBase::Static(..) => return false,
-            PlaceBase::Local(..) => {},
-        }
+    match place.base {
+        PlaceBase::Static(_) => false,
 
-        for proj in place_projection {
-            // Reborrow of already borrowed data is ignored
-            // Any errors will be caught on the initial borrow
-            if proj.elem == ProjectionElem::Deref {
-                return false;
-            }
-        }
+        // Reborrow of already borrowed data is ignored
+        // Any errors will be caught on the initial borrow
+        _ if place.is_indirect() => false,
 
-        true
-    })
+        PlaceBase::Local(_) => true,
```

Comment:
What do you think about changing the `true` to `!place.is_indirect()`?

Reply:
That works better.